internal/action/command.go

```diff
@@ -698,16 +698,29 @@ func (h *BufPane) QuitCmd(args []string) {
 }
 
 func convertLine(h *BufPane, line string) (int, error) {
+	fromEnd := false
+
+	// Check for special negative movement beginning from the end of the file
+	if strings.Contains(line, "~") {
+		line = strings.ReplaceAll(line, "~", "-")
```

Comment:
Why not `line = line[1:]`?

And generally IMO the code becomes a bit messy (this extra `fromEnd` state we need to remember below, etc). I would prefer:

```
	// Check for special negative movement beginning from the end of the file
	if strings.Contains(line, "~") {
		lineNum, err := strconv.Atoi(line[1:])
		if err != nil {
			return 0, err
		}
		lineNum = h.Buf.LinesNum() + 1 - lineNum
	} else {
		lineNum, err := strconv.Atoi(line)
		if err != nil {
			return 0, err
		}

		// Check for relative numbers
		if strings.HasPrefix(line, "-") || strings.HasPrefix(line, "+") {
			lineNum = h.Buf.GetActiveCursor().Y + 1 + lineNum
		}
	}

	lineNum = util.Clamp(lineNum-1, 0, h.Buf.LinesNum()-1)

Reply:
Ok, I got why you used replace. This trick IMO makes the code even messier, so I'd suggest my above version instead.